eclipse-collections-code-generator/src/main/resources/impl/list/immutable/immutablePrimitiveArrayList.stg

```diff
@@ -92,6 +92,12 @@ final class Immutable<name>ArrayList
         return new Immutable<name>ArrayList(newArray);
     }
 
+    public static Immutable<name>ArrayList newListFrom(<name>List itemList)
+    {
+        <type>[] newArray = itemList.toArray(new <type>[itemList.size()]);
+        return new Immutable<name>ArrayList(newArray);
+    }
```

Comment:
Is this an unrelated change that snuck in?

Reply:
It's actually the reason for the change I believe.

Reply:
This could be clarified in the commit message or be a separate commit to split the use case from the change.

Reply:
@motlin - a related change that snuck in. I will take it out from this PR.  @donraab is correct, there will be a separate change to take advantage of the toArray(target) in factory methods.